internal/controller/supervisorconfig/githubupstreamwatcher/github_upstream_watcher_test.go

```diff
@@ -6,74 +6,1205 @@ package githubupstreamwatcher
+			Message:            "allowAuthentication policy for organizations is valid",
+		}
+	}
+
+	buildOrganizationsPolicyValidFalse := func(t *testing.T, message string) metav1.Condition {
+		t.Helper()
+
+		return metav1.Condition{
+			Type:               OrganizationsPolicyValid,
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: generation,
+			LastTransitionTime: lastTransitionTime,
+			Reason:             "Invalid",
+			Message:            message,
+		}
+	}
+
+	buildClientCredentialsObtainedTrue := func(t *testing.T) metav1.Condition {
+		t.Helper()
+		return metav1.Condition{
+			Type:               ClientCredentialsObtained,
+			Status:             metav1.ConditionTrue,
+			ObservedGeneration: generation,
+			LastTransitionTime: lastTransitionTime,
+			Reason:             upstreamwatchers.ReasonSuccess,
+			Message:            "client credentials have been obtained, although they may not yet have been validated",
+		}
+	}
 
-	testNamespace := "foo"
-	testName := "bar"
+	buildClientCredentialsObtainedFalse := func(t *testing.T, secretName, namespace, reason string) metav1.Condition {
+		t.Helper()
+
+		return metav1.Condition{
+			Type:               ClientCredentialsObtained,
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: generation,
+			LastTransitionTime: lastTransitionTime,
+			Reason:             reason,
+			Message:            fmt.Sprintf(`secret with name %q must be found in namespace %q with type "secrets.pinniped.dev/github-client" and keys "clientID" and "clientSecret"`, secretName, namespace),
+		}
+	}
+
+	buildGitHubConnectionValidTrue := func(t *testing.T, host string) metav1.Condition {
+		t.Helper()
+
+		return metav1.Condition{
+			Type:               GitHubConnectionValid,
+			Status:             metav1.ConditionTrue,
+			ObservedGeneration: generation,
+			LastTransitionTime: lastTransitionTime,
+			Reason:             upstreamwatchers.ReasonSuccess,
+			Message:            fmt.Sprintf("spec.githubAPI.host (%q) is reachable and TLS verification succeeds", host),
+		}
+	}
+
+	buildGitHubConnectionValidUnknown := func(t *testing.T) metav1.Condition {
+		t.Helper()
+
+		return metav1.Condition{
+			Type:               GitHubConnectionValid,
+			Status:             metav1.ConditionUnknown,
+			ObservedGeneration: generation,
+			LastTransitionTime: lastTransitionTime,
+			Reason:             "UnableToValidate",
+			Message:            "unable to validate; see other conditions for details",
+		}
+	}
 
 	tests := []struct {
 		name                    string
 		githubIdentityProviders []runtime.Object
-		inputSecrets            []runtime.Object
+		secrets                 []runtime.Object
 		configClient            func(*pinnipedfake.Clientset)
-		wantErr                 string
+		wantErrRegexp           string
 		wantLogs                []string
-		wantResultingCache      []*githubtestutil.TestUpstreamGithubIdentityProvider
+		wantResultingCache      []*upstreamgithub.ProviderConfig
```

Comment:
So its a little painful, but I believe our pattern is to use `wantActions`, were we not only provide the `upstreams`, but also the `kubernetes API action` that wraps the object being sent.  There are lots of examples of this in our other controller tests.

See:
- 
- https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/jwtcachefiller/jwtcachefiller_test.go#L594
- https://github.com/vmware-tanzu/pinniped/blob/main/internal/controller/authenticator/webhookcachefiller/webhookcachefiller_test.go#L389